Bottom: 7aa77254eb0ba39d120f7606de6f659c9a43af58
Top:    f3f659704819015fd8ae495845d650fa0b80fe41
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:26:00 +0000

Add wrapper function for appending to GList of step_ids stuffed into pointers.

Avoids casting


---

diff --git a/src/hid/step/step_writer.c b/src/hid/step/step_writer.c
index 7f74349..9655d8f 100644
--- a/src/hid/step/step_writer.c
+++ b/src/hid/step/step_writer.c
@@ -34,6 +34,12 @@ destroy_step_id_list (step_id_list list)
   g_list_free (list);
 }
 
+step_id_list
+step_id_list_append (step_id_list list, step_id id)
+{
+  return g_list_append (list, GINT_TO_POINTER (id));
+}
+
 /* NB: The GList this produces will leak if not passed back to one of the step_* functions which uses the list and destroys it */
 step_id_list
 make_step_id_list (int count, ...)
diff --git a/src/hid/step/step_writer.h b/src/hid/step/step_writer.h
index 1490bb4..c86aa1e 100644
--- a/src/hid/step/step_writer.h
+++ b/src/hid/step/step_writer.h
@@ -8,6 +8,7 @@ typedef struct {
 
 } step_file;
 
+step_id_list step_id_list_append (step_id_list list, step_id id);
 step_id_list make_step_id_list (int count, ...);
 
 step_file *step_output_file (FILE *f);
